compiler/ccgtypes.nim

```diff
@@ -123,10 +123,11 @@ const
 
 proc typeName(typ: PType): Rope =
   let typ = typ.skipTypes(irrelevantForBackend)
-  result = if typ.sym != nil and typ.kind in {tyObject, tyEnum}:
-             typ.sym.name.s.mangle.rope
-           else:
-             ~"TY"
+  result =
+    if typ.sym != nil and typ.kind in {tyObject, tyEnum}:
+      typ.sym.name.s.mangle.rope
+    else:
+      rope($typ.kind)
```

Comment:
the result of ``$typ.kind`` already starts with "ty". I had what you suggested initially, but I thought it was stupid that all types start with TYty... Btw this implementation has no problems.